### DISCUSSION TITLE
Forward CloudTrail logs to Datadog

### DISCUSSION THREAD

**Discussion by rhoboat** [gruntwork-io/knowledge-base#179] [Help]
Hello!
We currently have a cloudtrail bucket set up in AWS using the cloudtrail-bucket module (https://github.com/gruntwork-io/terraform-aws-security/tree/master/modules/cloudtrail-bucket).

We are planning to introduce a Datadog Forwarder to our account to forward the CloudTrail logs to datadog. (https://docs.datadoghq.com/serverless/libraries_integrations/forwarder/)

It doesn't look like there are hooks in place to add additional policy statements to allow the forwarder to forward these logs. Is there a good method for accomplishing that that I am missing in the module?

If not - is there a recommended migration we could perform to different bucket module which would give us this flexibility?

**Comment by rhoboat** [comment]
I think the simplest solution would be to allow you to modify the existing CloudTrail S3 bucket to have permissions to forward objects to DataDog. I'll open an issue to see if we can get this prioritized so that you can do that.

In the meantime, the only other way I can see is that you'd use the private-s3-bucket module directly, along with custom-iam-entity or write your own policies. Then essentially you'd have CloudTrail write to the existing bucket and that bucket would copy contents into the new one you created, but the new one would also allow forwarding to DataDog. I think the best way is to allow you to plug your custom permissions into the existing bucket.

[Created the issue to track this feature here](https://github.com/gruntwork-io/terraform-aws-security/issues/609).

_help from @yorinasub17_

**Reply by beyjor** [reply]
A prioritized issue sounds great, thanks!

I agree that sounds like a good solution.

**Reply by beyjor** [reply]
I just realized we may have the same issue with our flow logs buckets which are driven by [vpc-flow-logs module](https://github.com/gruntwork-io/terraform-aws-vpc/tree/master/modules/vpc-flow-logs).

Any chance you'd consider adding the same capability there as well?
